odbc/src/api/functions.rs

```diff
@@ -1454,6 +1456,14 @@ unsafe fn sql_get_connect_attrw_helper(
                 let login_timeout = attributes.login_timeout.unwrap_or(0);
                 i32_len::set_output_fixed_data(&login_timeout, value_ptr, string_length_ptr)
             }
+            // according to the spec, SQL_ATTR_CONNECTION_DEAD just returns the latest status of the connection, not the current status
+            ConnectionAttribute::SQL_ATTR_CONNECTION_DEAD => {
+                let connection_dead = match *conn.mongo_connection.write().unwrap() {
+                    Some(_) => SqlBool::SQL_FALSE,
+                    None => SqlBool::SQL_TRUE,
+                };
```

Comment:
```suggestion
            ConnectionAttribute::SQL_ATTR_CONNECTION_DEAD => {
                let connection_dead = match *conn.mongo_connection.read().unwrap() {
                    Some(_) => SqlBool::SQL_FALSE,
                    None => SqlBool::SQL_TRUE,
                };
```
I don't think a write is necessary here

Reply:
good catch. I initially was pinging the database with the connection (hence the write) but realized that was more then the spec was asking for